internal/domain/power-status.go

```diff
@@ -18,16 +19,707 @@
+// distLockMaxTimeIn: Max time to hold dist'd lock before forfeiting
+// loggerIn:          Logrus logger object.  Will create if nil.
+// sampleInterval:    Time between HW status sampling.
+// Return:            nil on success, else error message.
+
+func PowerStatusMonitorInit(domGlb *DOMAIN_GLOBALS,
+                            distLockMaxTimeIn time.Duration,
+                            loggerIn *logrus.Logger,
+                            sampleInterval time.Duration) error {
+	if (loggerIn == nil) {
+		glogger = logrus.New()
+	} else {
+		glogger = loggerIn
+	}
+
+	if (domGlb.HSM == nil) {
+		return fmt.Errorf("ERROR: must supply valid HSM handle.")
+	}
+	if (domGlb.DSP == nil) {
+		return fmt.Errorf("ERROR: must supply valid storage object.")
+	}
+	if (domGlb.CS == nil) {
+		if (domGlb.VaultEnabled) {
+			return fmt.Errorf("ERROR: must supply valid component cred store handle.")
+		}
+	}
+	if (domGlb.DistLock == nil) {
+		return fmt.Errorf("ERROR: must supply valid distributed lock object.")
+	}
+	if (distLockMaxTimeIn < time.Second) {
+		return fmt.Errorf("ERROR: distributed lock max time must be >= 1 second.")
+	}
+	if (domGlb.RFTloc == nil) {
+		return fmt.Errorf("ERROR: must supply valid Task Runner Service handle.")
+	}
+	if (sampleInterval < time.Second) {
+		return fmt.Errorf("ERROR: power monitor sample interval must be >= 1 second.")
+	}
+
+	hsmHandle = domGlb.HSM
+	kvStore = domGlb.DSP
+	ccStore = domGlb.CS
+	distLocker = domGlb.DistLock
+	tloc = domGlb.RFTloc
+	distLockMaxTime = distLockMaxTimeIn
+	pmSampleInterval = sampleInterval
+	serviceRunning = domGlb.Running
+	vaultEnabled = domGlb.VaultEnabled
+
+	go monitorHW()
+	return nil
+}
+
+// Stop power status monitoring.
+
+func PowerStatusMonitorStop() {
+	pstateMonitorRunning = false
+}
+
+// Change the power status monitoring interval, on the fly.
+
+func PowerStatusMonitorChangeInterval(newInterval time.Duration) error {
+	if (newInterval < time.Second) {
+		return fmt.Errorf("ERROR: power monitor sample interval must be >= 1 second.")
+	}
+	pmSampleInterval = newInterval
+	return nil
+}
+
+// Get power status for given components.  Filter by power state and 
+// management state.  Any undefined filter results in all states for
+// the state category.
+//
+// xnames:          Array of component names for which to get power data
+// pwrStateFilter:  Power state filter.
+// mgmtStateFilter: Management state filter.  
+// Return:          Passback object populated with model.PowerStatus object.
+
+func GetPowerStatus(xnames []string,
+                    pwrStateFilter pcsmodel.PowerStateFilter,
+                    mgmtStateFilter pcsmodel.ManagementStateFilter) (pb pcsmodel.Passback) {
+	//Grab all KVs from ETCD, cycle through them to make a map, then
+	//match the xnames in the passed-in array.  Grab pertinent data and create
+	//a pcsmodel.PowerStatus object 'pstatus', and return it.
+
+	statusObj,err := (*kvStore).GetAllPowerStatus()
```

Comment:
Will `GET /power-status` just return nothing if power monitoring is not started?

Reply:
You would get an empty status[] array.